src/components/App/App.component.jsx

```diff
@@ -1,58 +1,71 @@
-import Fortune from '../Fortune';
-import Layout from '../Layout';
-import { random } from '../../utils/fns';
-
-function App() {
-  useLayoutEffect(() => {
-    const { body } = document;
-
-    function rotateBackground() {
-      const xPercent = random(100);
-      const yPercent = random(100);
-      body.style.setProperty('--bg-position', `${xPercent}% ${yPercent}%`);
-    }
-
-    const intervalId = setInterval(rotateBackground, 3000);
-    body.addEventListener('click', rotateBackground);
+import React, { useEffect, useContext } from 'react';
+import { Switch, Route, Redirect, useLocation } from 'react-router-dom';
+
+import { AuthContext } from '../../providers/auth';
+import { setCurrentAuth } from '../../providers/auth/auth.actions';
+
+import HomePage from '../../pages/home';
+import WatchVideoPage from '../../pages/watch-video';
+import SettingsPage from '../../pages/settings';
+import FavoritesPage from '../../pages/favorites';
+import LogInPage from '../../pages/log-in';
+import SignUpPage from '../../pages/sign-up';
+
+import Header from '../header';
+
+import { auth, createUserProfileDocument } from '../../utils/js/firebase';
+
+import { AppContainer, AppMainContainer } from './app.styles';
+
+const App = () => {
+  const location = useLocation();
+  const { authState, authDispatch } = useContext(AuthContext);
+  const { currentAuth } = authState;
+
+  useEffect(() => {
+    const unsubscribeFromAuth = auth.onAuthStateChanged(async (userAuth) => {
+      if (userAuth) {
+        const userRef = await createUserProfileDocument(userAuth);
+
+        userRef.onSnapshot((snapShot) => {
+          const newState = {
+            id: snapShot.id,
+            ...snapShot.data(),
+          };
+
+          authDispatch(setCurrentAuth(newState));
+        });
+      } else {
+        authDispatch(setCurrentAuth(null));
+      }
+    });
 
     return () => {
-      clearInterval(intervalId);
-      body.removeEventListener('click', rotateBackground);
+      unsubscribeFromAuth();
     };
-  }, []);
+  }, [authDispatch]);
 
   return (
-    <BrowserRouter>
-      <AuthProvider>
-        <Layout>
-          <Switch>
-            <Route exact path="/">
-              <HomePage />
-            </Route>
-            <Route exact path="/login">
-              <LoginPage />
-            </Route>
-            <Private exact path="/secret">
-              <SecretPage />
-            </Private>
-            <Route path="*">
-              <NotFound />
-            </Route>
-          </Switch>
-          <Fortune />
-        </Layout>
-      </AuthProvider>
-    </BrowserRouter>
+    <AppContainer>
+      {location.pathname !== '/login' && location.pathname !== '/signup' ? (
+        <Header />
+      ) : null}
+      <AppMainContainer>
+        <Switch>
+          <Route exact path='/' component={HomePage} />
+          <Route path='/settings' component={SettingsPage} />
+          <Route exact path='/favorites' component={FavoritesPage} />
```

Comment:
Just a comment in case you missed it, the favorite page should be only accessible when the user is authenticated